File: client-app/src/app/models/material.ts
export interface Material {
    id: string
    name: string
    price: number
    description: string
    date: Date | null
    quantity?: number | null;
    weight?: number | null;
    quantityRequired?: number | null;
    weightRequired?: number | null;
    country: string
    factory: string
  }

File: Domain/Material.cs


namespace Domain
{
    public class Material
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int? Quantity { get; set; }
        public int? QuantityRequired { get; set; }
        public double? Weight { get; set; }
        public double? WeightRequired { get; set; }
        public double Price { get; set; }
        public DateTime Date {get; set;}
        public string Country { get; set; }
        public string Factory { get; set; }
        public bool IsManufactoring { get; set; }
        public List<MaterialProduct> MaterialProducts { get; set; }
    }
}